parquet/src/test/java/org/apache/iceberg/parquet/TestParquet.java

```diff
@@ -97,15 +118,14 @@ public void testNumberOfBytesWritten() throws IOException {
     Assert.assertEquals(expectedSize, actualSize);
   }
 
-  private Pair<File, Long> generateFileWithTwoRowGroups(Function<MessageType, ParquetValueWriter<?>> createWriterFunc)
+  private Pair<File, Long>  generateFileWithTwoRowGroups(
```

Comment:
The modification to these tests doesn't seem to fit. This is pulling the writer configuration out of this method, but still using the name `generateFileWithTwoRowGroups` even though there's no longer a guarantee that will actually happen. I think it would be better to build the properties in this method and pass in settings:

```java
private ... generateFile(Function<...> createWriterFunc, Long rowGroupSizeBytes, Integer minCheckRecordCount, Integer maxCheckRecordCount)
```

That way, all this function does is create the file. I think it should also set its own desired record count based on the min row group count so you don't have to pass in 1 more than the row group size in records.

Reply:
Updated the method to build the props inside this method.

`desiredRecordCount` is already part of the method signature. No more hardcoded 1 more than the row group size in records.